### ISSUE TITLE
User with Project Owner role cannot use ConfigMap/Secrets for environment variable in deployments

### ISSUE THREAD

**Issue by sgapanovich** [rancher/dashboard#4976]
issue: SURE-3916

**Setup**
- Rancher version: 2.6-head 4763f8d
- Browser type & version: Chrome 

**Describe the bug**

A user with Project Owner role can't use Config Maps or Secrets when creating a deployment. The dropdown shows "Sorry, no matching options." even though resources are returned by APIs and the user can use those config maps and secrets when creating deployments though backend (for ex: via `kubectl`)

**To Reproduce**

Can be reproduced in a local cluster but since a downstream
1. Crete a downstream cluster
2. Create a standard user (user 1)
3. Add the user to the default project in the cluster 
  1.Go to the cluster
  2.Project/Namespaces
  3."Edit Config" for the Default Project
  4."Add"
  5.Find the user you created
  6.Select "Owner" as their role
4. Log in to Rancher as the user 1
  The user should be able to see the cluster and have access to the default project/namespace
5. Go to Storage > ConfigMaps > Create > enter any values for key and value
6. Go to Workload > Deployments > Create > Find "Environment Variables" section and click "Add Variable" > type: Config Map > Click on the Config Map dropdown

**Result**
user sees "Sorry, no matching options."

**Expected Result**

User should be able to see Config Maps from default namespace since the user is the project owner of the default project and their namespaces

That's not expected since the user has access to those config maps if they go to view config maps on UI or use `kubectl get cm`
On the create workload page the backend returns the config map data in `GET https://URL/k8s/clusters/cluster-name/v1/configmaps` but UI does not display that 

The user is able to create a deployment using the config maps from default namespace when they use `kubectl` or execute API calls to the backend

In the original issue there is a mention of "if a user has a permission/role to `get nodes` the user will be able to view the config maps" -> i tested it and this workaround worked in the local cluster but not in the downstream cluster.

Note: the same for secrets 

**Screenshots**

![repro](https://user-images.githubusercontent.com/40282141/150654596-e554945e-bd22-471a-9287-d9cfa52ba92f.jpg)

**Comment by AlexisDucastel** [comment]
Hi !

Exactly the same problem here ! Thanks.

(Rancher version 2.6.3)

**Comment by samjustus** [comment]
SURE-4154

**Comment by catherineluse** [comment]
The secrets part of this issue may have already been fixed https://github.com/rancher/dashboard/issues/5084

**Comment by nwmac** [comment]
@mantis-toboggan-md Validate - possibly aleady fixed

**Comment by brudnak** [comment]
# :white_check_mark: PASSED

## Reproduction Environment

| Component                 | Version / Type                           |
| ------------------------- | ---------------------------------------- |
| Rancher version           | 2.6.4                                    |
| Rancher commit link       | not applicable                           |
| Installation option       | helm high availability                   |
| RKE binary version used   | v1.3.8                                   |
| If Helm Chart k8s cluster | v1.22.7                                  |
| Cert Details              | let's encrypt                            |
| Docker version            | 20.10.7, build f0df350                   |
| Helm version              | v2.16.8-rancher1                         |
| Downstream cluster type   | rke1 linode                              |
| Downstream K8s version    | v1.22.7-rancher1-2                       |
| Logged in user role       | administrator & standard user            |
| Browser type              | google chrome                            |
| Browser version           | 100.0.4896.127 (official build) (x86_64) |

**ADDITIONAL SETUP**

- Helm commands for Rancher installation:

```shell
kubectl create namespace cattle-system
```

```shell
helm install cert-manager jetstack/cert-manager \
  --namespace cert-manager \
  --create-namespace \
  --version v1.7.1 \
  --set installCRDs=true
```

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$OMITTED \
  --set bootstrapPassword=admin \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$OMITTED \
  --set letsEncrypt.ingress.class=nginx \
  --version 2.6.4
```

## Reproduction steps

1. Crete a downstream cluster
2. Create a standard user (user 1)
3. Add the user to the default project in the cluster
  1. Go to the cluster
  2. Project/Namespaces
  3. "Edit Config" for the Default Project
  4. "Add"
  5. Find the user you created
  6. Select "Owner" as their role
4. Log in to Rancher as the user 1
5. The user should be able to see the cluster and have access to the default project/namespace
6. Go to Storage > ConfigMaps > Create > enter any values for key and value
7. Go to Workload > Deployments > Create > Find "Environment Variables" section and click "Add Variable" > type: Config Map > Click on the Config Map dropdown
  
## Additional Info
>Issue resolved in Rancher 2.6.4

## RESULTS

#### :white_check_mark: Expected
Config map is accessible

#### :white_check_mark: Actual
Config map is accessible

<hr>
  
## Validation Environment

| Component                 | Version / Type                           |
| ------------------------- | ---------------------------------------- |
| Rancher version           | 2.6.5-rc2                                |
| Rancher commit link       | not applicable                           |
| Installation option       | helm high availability                   |
| RKE binary version used   | v1.3.8                                   |
| If Helm Chart k8s cluster | v1.22.7                                  |
| Cert Details              | let's encrypt                            |
| Docker version            | 20.10.7, build f0df350                   |
| Helm version              | v2.16.8-rancher1                         |
| Downstream cluster type   | rke1 linode                              |
| Downstream K8s version    | v1.22.8-rancher1-1                       |
| Logged in user role       | administrator & standard user            |
| Browser type              | google chrome                            |
| Browser version           | 100.0.4896.127 (official build) (x86_64) |

**ADDITIONAL SETUP**

- Helm commands for Rancher installation:

```shell
kubectl create namespace cattle-system
```

```shell
helm install cert-manager jetstack/cert-manager \
  --namespace cert-manager \
  --create-namespace \
  --version v1.7.1 \
  --set installCRDs=true
```

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$OMITTED \
  --set bootstrapPassword=admin \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$OMITTED \
  --set letsEncrypt.ingress.class=nginx \
  --version 2.6.5-rc2
```

## Validation steps

1. Crete a downstream cluster
2. Create a standard user (user 1)
3. Add the user to the default project in the cluster
  1. Go to the cluster
  2. Project/Namespaces
  3. "Edit Config" for the Default Project
  4. "Add"
  5. Find the user you created
  6. Select "Owner" as their role
4. Log in to Rancher as the user 1
5. The user should be able to see the cluster and have access to the default project/namespace
6. Go to Storage > ConfigMaps > Create > enter any values for key and value
7. Go to Workload > Deployments > Create > Find "Environment Variables" section and click "Add Variable" > type: Config Map > Click on the Config Map dropdown
  
## Additional Info

## RESULTS

#### :white_check_mark: Expected
Config map is accessible

#### :white_check_mark:Actual
Config map is accessible
